parlai/core/metrics.py

```diff
@@ -177,27 +185,29 @@ def _rouge(guess, answers):
         )
         rouge = None
         return None, None, None
-
-    scores_rouge1 = [score['rouge-1']['r'] for score in scores]
-    scores_rouge2 = [score['rouge-2']['r'] for score in scores]
-    scores_rougel = [score['rouge-l']['r'] for score in scores]
-    return max(scores_rouge1), max(scores_rouge2), max(scores_rougel)
+    scores_rouge_n = []
+    for idx in range(1, rougen):
+        scores_rouge_n.append(max(score['rouge-{}'.format(idx)]['r'] for score in scores))
+    scores_rouge_n = scores_rouge_n if rougen else None
+    scores_rougel = max([score['rouge-l']['r'] for score in scores]) if rougel else None
+    return scores_rouge_n, scores_rougel
 
 
 def aggregate_metrics(reporters):
     """Aggregate metrics from multiple reports."""
     # reporters is a list of teachers or worlds
     m = {}
     m['tasks'] = {}
-    sums = {'accuracy': 0, 'f1': 0, 'loss': 0, 'ppl': 0}
-    if nltkbleu is not None:
-        sums['bleu'] = 0
-    if rouge is not None:
-        sums['rouge-1'] = 0.0
-        sums['rouge-2'] = 0.0
-        sums['rouge-L'] = 0.0
+    sums = {}
     num_tasks = 0
     total = 0
+    for i in range(len(reporters)):
+        task_id = reporters[i].getID()
+        task_report = reporters[i].report()
+        for each_metric, value in task_report.items():
+            if isinstance(value, Number):
+                sums[each_metric] = 0.0
```

Comment:
you should respect ints/floats here. There's some nuance (floats get rounded by round_sigfigs later on, ints don't).